src/client/components/Inputs/InputNumber/InputNumber.tsx

```diff
@@ -0,0 +1,85 @@
+import './InputNumber.scss'
+import React, { forwardRef, useEffect, useImperativeHandle, useRef, useState } from 'react'
+
+import classNames from 'classnames'
+import { Numbers } from 'utils/numbers'
+import { Objects } from 'utils/objects'
+
+import { useOnBlur } from './hooks/useOnBlur'
+import { useOnChange } from './hooks/useOnChange'
+import { InputNumberProps } from './types'
+
+const InputNumber = forwardRef<HTMLInputElement, InputNumberProps>((props, outerRef) => {
+  const {
+    disabled,
+    id,
+    isInteger,
+    maxLength,
+    onChange,
+    onPaste,
+    placeholder,
+    precision,
+    shouldRound,
+    thousandSeparated,
+    value,
+  } = props
+
+  const inputRef = useRef<HTMLInputElement>(null)
+  useImperativeHandle(outerRef, () => inputRef.current!, [])
+
+  const [localValue, setLocalValue] = useState<typeof value>(value)
```

Comment:
When removing the local value, this happens when the `value` from the caller is async and formatted/sanitized:

https://github.com/user-attachments/assets/6fe4d055-1073-4428-9416-c672608ffb91

Issues in this vid:
- When you start typing you get the .00 format. 
- You can experience that you delete something but it comes back for a split second because of the async+formatting.
- Deleting from the end of the input is funky. If you go from, let's say `123.00` to `123.0` while deleting, the format applied by the component caller is two decimals so you go back to `123.00`. 

It's similar to the issues we had with the EditorWYSIWYG, since the value is sanitized/formatted and async. There the fix was to call the passed `onChange` only when the editor is out of focus.

We could remove the local value and some other logic but then this behavior would need to be handled by the caller. What do you think?

Reply:
The core issue is separating when the formatting is applied from when the user is typing. We don't have these problems with the `InputText` because we don't format/sanitize the values where we use it.

Reply:
Yes, I mean the  format applied to the `value` prop by the caller `onChange`. in this case, when updating the node value: 
<img width="719" alt="image" src="https://github.com/user-attachments/assets/3d2198d3-4939-4fde-8b95-880bf69f949b">
At `src/client/pages/Section/DataTable/Table/RowData/Cell/hooks/useOnChange.ts` 
`Sanitizer.sanitize` also formats the value, which is why if you delete a 0 from `123.00` it stays as `123.00`. 
I could remove that for decimal and integer value types. 

Also, if we don't have local value, the caller needs to handle the input of the decimal period, eg `123.` would need to be considered valid so you can get to input `123.1` using the value prop. At the moment, `Sanitizer.isAcceptable` would return false for `123.`so the value update wouldn't run and the input would get stuck at `123`